packages/twenty-front/src/modules/object-record/record-group/hooks/useRecordGroupSelector.ts

```diff
@@ -0,0 +1,110 @@
+import { FieldMetadataItem } from '@/object-metadata/types/FieldMetadataItem';
+import { useRecoilComponentCallbackStateV2 } from '@/ui/utilities/state/component-state/hooks/useRecoilComponentCallbackStateV2';
+import { usePersistViewGroupRecords } from '@/views/hooks/internal/usePersistViewGroupRecords';
+import { useGetViewFromCache } from '@/views/hooks/useGetViewFromCache';
+import { currentViewIdComponentState } from '@/views/states/currentViewIdComponentState';
+import { ViewGroup } from '@/views/types/ViewGroup';
+import { useRecoilCallback } from 'recoil';
+import { v4 } from 'uuid';
+import { isUndefinedOrNull } from '~/utils/isUndefinedOrNull';
+
+type UseRecordGroupSelectorParams = {
+  viewBarComponentId: string;
+};
+
+export const useRecordGroupSelector = ({
+  viewBarComponentId,
+}: UseRecordGroupSelectorParams) => {
+  const { createViewGroupRecords, deleteViewGroupRecords } =
+    usePersistViewGroupRecords();
+
+  const currentViewIdCallbackState = useRecoilComponentCallbackStateV2(
+    currentViewIdComponentState,
+    viewBarComponentId,
+  );
+
+  const { getViewFromCache } = useGetViewFromCache();
```

Comment:
we might want to also split this hook in pieces as there is no need for sync functions (recoil callback) to be re-rendered when getVie